kzg_prover/examples/chunked_univariate_grand_sum.rs

```diff
@@ -0,0 +1,183 @@
+
+    const BALANCES_INDEX: usize = 1;
+    // Get the 1st advice polynomial (user balances #0) from each chunk
+    let f_poly_1 = advice_polys_1.advice_polys.get(BALANCES_INDEX).unwrap();
+    let f_poly_2 = advice_polys_2.advice_polys.get(BALANCES_INDEX).unwrap();
+
+    // These advice polynomials can then be used to independently produce the user inclusion KZG proofs.
+    // This allows to significantly speed up the inclusion proof by using smaller `N_USERS_CHUNK` size
+    // and parallelizing the proof generation.
+
+    // Take the KZG commitment of each chunk from the zkSNARK proof transcript
+    let mut transcript_1 = Keccak256Transcript::new(proof_1.as_slice());
+    let mut advice_commitments_1 = Vec::new();
+    (0..N_CURRENCIES + 1).for_each(|_| {
+        let point: G1Affine = transcript_1.read_point().unwrap();
+        advice_commitments_1.push(point);
+    });
+    let kzg_commitment_1 = advice_commitments_1[BALANCES_INDEX];
+    let mut transcript_2 = Keccak256Transcript::new(proof_2.as_slice());
+    let mut advice_commitments_2 = Vec::new();
+    (0..N_CURRENCIES + 1).for_each(|_| {
+        let point: G1Affine = transcript_2.read_point().unwrap();
+        advice_commitments_2.push(point);
+    });
+    let kzg_commitment_2 = advice_commitments_2[BALANCES_INDEX];
+    assert!(
+        kzg_commitment_1 != kzg_commitment_2,
+        "Commitments should be different"
+    );
+
+    // The homomorphic property of KZG commitments allows us to sum the individual chunk commitments
+    // to produce the KZG opening proof for the grand total
+    let kzg_commitment_sum = kzg_commitment_1 + kzg_commitment_2;
+
+    // First, add the polynomials together coefficient-wise
+    let domain = vk.get_domain();
+    let mut f_poly_total = domain.empty_coeff();
+
+    for (poly, value) in f_poly_total
+        .iter_mut()
+        .zip(f_poly_1.iter().zip(f_poly_2.iter()))
+    {
+        *poly = *value.0 + *value.1;
+    }
+
+    // Demonstrating the homomorphic property of KZG commitments. The sum of the KZG commitments
+    // to the chunk polynomials should be the same as the KZG commitment to the total polynomial
+    // that is a sum of the chunk polynomials
+    let kzg_commitment_total = commit_kzg(&params, &f_poly_total);
+    assert!(
+        kzg_commitment_sum.to_affine() == kzg_commitment_total.to_affine(),
+        "Commitments should be equal"
+    );
+
+    let poly_length = 1 << u64::from(K);
+
+    // We're opening the resulting polynomial at x = 0 and expect the constant coefficient
+    // to be equal to the grand total divided by the size of the polynomial
+    // thanks to the univariate grand sum property.
+    let challenge = Fp::ZERO;
+    let kzg_proof = create_naive_kzg_proof::<KZGCommitmentScheme<Bn256>>(
+        &params,
+        &domain,
+        &f_poly_total,
+        challenge,
+        big_uint_to_fp(&(csv_total[BALANCES_INDEX - 1])) * Fp::from(poly_length).invert().unwrap(),
```

Comment:
Could you assign this to a variable and reuse it in the `verify_kzg_proof` call?
This change could improve code readability.